sunpy/net/dataretriever/client.py

```diff
@@ -206,17 +215,47 @@ def get(self, qres, **kwargs):
         -------
         Results Object
         """
+        details = (qres.__dict__['client']).map_
+
         urls = []
         for qrblock in qres:
             urls.append(qrblock.url)
 
+        filenames = []
+        for url in urls:
+            filenames.append(url.split('/')[-1])
+
+        # Create function to compute the filepath to download to if not set
+        default_dir = sunpy.config.get("downloads", "download_dir")
+
+        paths = []
+        for filename in filenames:
+            if path is None:
+                path = os.path.join(default_dir, '{file}')
+            elif isinstance(path, six.string_types):
+                path = os.path.join(path, '{file}')
```

Comment:
you should check to see if `{file}` is in path first, otherwise if the user provides it, it could end up there twice.